ehsm_kms_service/test/test_kms_with_rest.py

```diff
@@ -28,66 +28,114 @@ def test_params(payload):
     return params
 
-def test_Stest_RSA3072_sign_verify(base_url, headers):
-    print('====================test_Stest_RSA3072_sign_verify start===========================')
+def test_RSA3072_encrypt_decrypt(base_url, headers):
+    print('====================test_RSA3072_encrypt_decrypt start===========================')
     params=test_params({
             "keyspec":"EH_RSA_3072",
             "origin":"EH_INTERNAL_KEY"
         })
     print('CreateKey req:\n%s\n' %(params))
-    create_resp = requests.post(url=base_url + "CreateKey", data=json.dumps(params), headers=headers)
-    print('CreateKey resp:\n%s\n' %(create_resp.text))
+    create_res = requests.post(url=base_url + "CreateKey", data=json.dumps(params), headers=headers)
+    print('CreateKey resp:\n%s\n' %(create_res.text))
+    if(check_result(create_res, 'CreateKey', 'test_RSA3072_encrypt_decrypt') == False):
+        return
+
+    # test AsymmetricEncrypt("123456")
+    params=test_params({
+            "cmk_base64": json.loads(create_res.text)['result']['cmk_base64'],
+            "plaintext": "123456"
+        })
+    print('AsymmetricEncrypt req:\n%s\n' %(params))
+    asymmetricEncrypt_res = requests.post(url=base_url + "AsymmetricEncrypt", data=json.dumps(params), headers=headers)
+    print('AsymmetricEncrypt resp:\n%s\n' %(asymmetricEncrypt_res.text))
+    if(check_result(asymmetricEncrypt_res, 'AsymmetricEncrypt', 'test_RSA3072_encrypt_decrypt') == False):
+        return
+    # test AsymmetricDecrypt(ciphertext)
+    params=test_params({
+            "cmk_base64": json.loads(create_res.text)['result']['cmk_base64'],
+            "ciphertext_base64": json.loads(asymmetricEncrypt_res.text)['result']['ciphertext_base64']
+        })
+    print('AsymmetricDecrypt req:\n%s\n' %(params))
+    asymmetricDecrypt_res = requests.post(url=base_url + "AsymmetricDecrypt", data=json.dumps(params), headers=headers)
+    print('AsymmetricDecrypt resp:\n%s\n' %(asymmetricDecrypt_res.text))
+    if(check_result(asymmetricDecrypt_res, 'AsymmetricDecrypt', 'test_RSA3072_encrypt_decrypt') == False):
+        return
+    plaintext = str(base64.b64decode(json.loads(asymmetricDecrypt_res.text)['result']['plaintext_base64']), 'utf-8').strip(b"\x00".decode())
+    print('AsymmetricDecrypt plaintext:\n%s\n' %(plaintext))
+    print('====================test_RSA3072_encrypt_decrypt end===========================')
+
+def test_Stest_RSA3072_sign_verify(base_url, headers):
+    print('====================test_Stest_RSA3072_sign_verify start===========================')
+    params=test_params({
+            "keyspec": "EH_RSA_3072",
+            "origin": "EH_INTERNAL_KEY"
+        })
+    print('CreateKey req:\n%s\n' %(params))
+    create_res = requests.post(url=base_url + "CreateKey", data=json.dumps(params), headers=headers)
+    print('CreateKey resp:\n%s\n' %(create_res.text))
+    if(check_result(create_res, 'CreateKey', 'test_Stest_RSA3072_sign_verify') == False):
```

Comment:
why not put the check_result() before print out the response?

exchange the line#75 and line#76, and same with below...

Reply:
It will be fix